core/templates/domain/story/editable-story-backend-api.service.spec.ts

```diff
@@ -118,134 +105,145 @@ describe('Editable story backend API service', function() {
       });
       expect(failHandler).not.toHaveBeenCalled();
     }
-  );
+    ));
 
   it('should successfully delete a story from the backend',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect(
-        'DELETE', '/story_editor_handler/data/storyId').respond(200);
-      EditableStoryBackendApiService.deleteStory('storyId').then(
+      editableStoryBackendApiService.deleteStory('storyId').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/story_editor_handler/data/storyId');
+      expect(req.request.method).toEqual('DELETE');
+      req.flush(200);
+
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalled();
       expect(failHandler).not.toHaveBeenCalled();
     }
-  );
+    ));
 
   it('should use the rejection handler if the backend request failed',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect(
-        'GET', '/story_editor_handler/data/2').respond(
-        500, 'Error loading story 2.');
-      EditableStoryBackendApiService.fetchStory('2').then(
+      editableStoryBackendApiService.fetchStory('2').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/story_editor_handler/data/2');
+      expect(req.request.method).toEqual('GET');
+      req.flush({status: 500, statusText:'Error loading story 2.'});
```

Comment:
All rejection req.flush(..) statements should be like this:
```
req.flush(<error message>, {
  status: <status code>,
  statusText: <error message>
});
```

Please change this here and below.

Reply:
Done.